### ISSUE TITLE
Tensorflow auto killed my training processing.

### ISSUE THREAD

**Issue by Runist** [tensorflow/tensorflow#44339]
**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow):
- OS Platform and Distribution (e.g., Linux Ubuntu 18.04):
- TensorFlow installed from (source or binary): binary
- TensorFlow version (use command below): tf-nightly-2.4.0
- Python version:
- Bazel version (if compiling from source): 3.7.4
- GCC/Compiler version (if compiling from source): 7.3.0
- CUDA/cuDNN version: CUDA 11.0 / cuDNN 8.0.4
- GPU model and memory: RTX3090 24GB
- RAM: 32GB

I'm training my Faster RCNN model.And I train it in my old GPU(1080 8GB).It didn't occur any error.Last week, I changed to RTX3090 and environment. When I run my code, I can run normal in first one hour. After that keras show the code runtime getting longer. I don't get any information, just get 'killed'.Someone saies  maybe high RAM usage, and I watch my memory, it auto increasing indeed. But I can't find code bug.
So I want to know if tf-nightly-gpu version bug?Or GPU environment setting?

Thanks~

**Comment by amahendrakar** [comment]
@Runist,
In order to expedite the trouble-shooting process, could you please provide the complete code to reproduce the issue reported here and also the output log. Thanks!

**Comment by Runist** [comment]
> @Runist,
> In order to expedite the trouble-shooting process, could you please provide the complete code to reproduce the issue reported here and also the output log. Thanks!

Yeah, this is my [code](https://github.com/Runist/Faster_RCNN).I am not sure this format is your want.

**Comment by Runist** [comment]
> @Runist,
> In order to expedite the trouble-shooting process, could you please provide the complete code to reproduce the issue reported here and also the output log. Thanks!

By the way, I didn't get any output log or information about my error.The only thing that happened was that the program was running slower and slower, each training step took longer and longer, the available memory was getting smaller and smaller until it was used up, and then "killed" was printed on the screen.

**Comment by Runist** [comment]
I tried to reduce GPU memory, but this error still exists.However I also test it in my DenseNet demo.It works normal.

**Comment by amahendrakar** [comment]
@Runist,
While trying to download the `faster_rcnn.h5` weights file I am facing a 404 error, please find the gist of it [here](https://colab.research.google.com/gist/amahendrakar/58ab1c9e00ca68a24845e93524531661/44339.ipynb). 

Also, try setting a hard limit on the total GPU memory as mentioned in [this guide](https://www.tensorflow.org/guide/gpu#limiting_gpu_memory_growth) and let us know if it helps. Thanks!

**Comment by Runist** [comment]
> @Runist,
> While trying to download the `faster_rcnn.h5` weights file I am facing a 404 error, please find the gist of it [here](https://colab.research.google.com/gist/amahendrakar/58ab1c9e00ca68a24845e93524531661/44339.ipynb).
> 
> Also, try setting a hard limit on the total GPU memory as mentioned in [this guide](https://www.tensorflow.org/guide/gpu#limiting_gpu_memory_growth) and let us know if it helps. Thanks!

Because the code could not work normal, I didn't get right model, I just reserved a spot.And I have try to set GPU memory to 7GB, it is same as RTX 1080, it also lead to RAM usage growth.
So I also change RTX 1080 to test, I want to know it is GPU problem or my code problem.Even RTX 1080 it also will lead to RAM usage growth.But RTX 1080 doesn't fill up memory quickly because of its small memory and slow computation.
Last week, I find the problem is "train_on_batch", I commented out all the code except for the TF.data input section.Just for loop this, it doesn't lead RAM usage growth.When I add "train_on_batch" and test.The RAM usage growth.
So maybe will help you.I updated my code once with using python generator input data.If you want to use tf.data  version, you should download the previous version.

**Comment by ymodak** [comment]
Apologies for the delay in response. Can you try with latest tf-nightly (2.5 version) or with recently released TF 2.4-rc2 version for cuda 11.0 support?

**Comment by Runist** [comment]
> Apologies for the delay in response. Is this still an issue? Can you try with latest tf-nightly (2.5 version) or with recently released TF 2.4-rc2 version for cuda 11.0 support?

I use rtx3090 by tf-nightly-2.5 and cuda 11.1. Cuda 11.0 can't work. I can avoid memory leaky via **tf.Gradient**.So I guarantee it was **train_on_batch**.And I test it in tf-1.13, it work normally.I also goole my problem, I find same [issue#33030](https://github.com/tensorflow/tensorflow/issues/33030).

**Comment by ymodak** [comment]
Thanks for sharing your investigation. Should we close this issue since you have found a workaround?

**Comment by Runist** [comment]
Casually, if you have more important to do you can close it, but I still think you should open it. BecauseI found that TF =2.0.0 train_on_batch existed memory leak, but now it is TF-nightly =2.5.0. There is still a problem.I just found a solution, but that doesn't mean the train_on_batch memory leak problem is solved.

**Comment by sachinprasadhs** [comment]
Could you please test it in Tensorflow 2.7 and let us know if the problem still exists. Thanks!
